scripts/end-to-end/extract-testcases.py

```diff
@@ -0,0 +1,284 @@
+#!/usr/bin/env python3
+# pylint: disable=eval-used, too-many-arguments, too-many-branches, too-many-nested-blocks, too-many-return-statements, too-many-instance-attributes
+
+import re
+import os
+
+
+class Test:
+    def __init__(self, name, content):
+        self.name = name
+        self.content = content
+        self.extractable = True
+        self.source = ""
+        self.comment = ""
+        self.tests = []
+        self.not_matching_lines = []
+        self.alsoViaYul = False
+
+    def analyse(self):
+        self.extractable = True
+        self.source = self.get_source()
+        self.content = self.content.replace(self.source, "", 1)
+
+        for not_allowed in ["m_contractAddress", "m_compiler", "m_evmHost",
```

Comment:
Wouldn't it be safer to have a positive list instead of a negative one, at least for the C++ identifiers?